From 433cd321bfbbeea5b4fcbf1a8674c406ce2f8681 Mon Sep 17 00:00:00 2001
From: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Date: Wed, 17 Nov 2021 12:17:07 +0100
Subject: [PATCH] usb: dwc2: move gadget resume out of
 dwc2_gadget_exit_clock_gating

This is a precursor patch to allow calling dwc2_gadget_exit_clock_gating()
from the platform suspend routine.
Calling gadget resume from there isn't suitable, so move it out of
dwc2_gadget_exit_clock_gating().

Signed-off-by: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Change-Id: Iec6e9d82e7d6dd80e79cf11270d3665f94488678
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/229474
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 drivers/usb/dwc2/core_intr.c | 3 ++-
 drivers/usb/dwc2/gadget.c    | 1 -
 2 files changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/usb/dwc2/core_intr.c
+++ b/drivers/usb/dwc2/core_intr.c
@@ -425,7 +425,6 @@ static void dwc2_handle_wakeup_detected_
 				if (ret)
 					dev_err(hsotg->dev,
 						"exit partial_power_down failed\n");
-				call_gadget(hsotg, resume);
 			}
 
 			/* Exit gadget mode clock gating. */
@@ -433,6 +432,8 @@ static void dwc2_handle_wakeup_detected_
 			    DWC2_POWER_DOWN_PARAM_NONE && hsotg->bus_suspended &&
 			    !hsotg->params.no_clock_gating)
 				dwc2_gadget_exit_clock_gating(hsotg, 0);
+
+			call_gadget(hsotg, resume);
 		} else {
 			/* Change to L0 state */
 			hsotg->lx_state = DWC2_L0;
--- a/drivers/usb/dwc2/gadget.c
+++ b/drivers/usb/dwc2/gadget.c
@@ -5680,7 +5680,6 @@ void dwc2_gadget_exit_clock_gating(struc
 	}
 
 	/* Change to L0 state */
-	call_gadget(hsotg, resume);
 	hsotg->lx_state = DWC2_L0;
 	hsotg->bus_suspended = false;
 }
